mne/io/pick.py

```diff
@@ -374,6 +372,10 @@ def pick_types(info, meg=True, eeg=False, stim=False, eog=False, ecg=False,
     """
     # NOTE: Changes to this function's signature should also be changed in
     # PickChannelsMixin
+    if meg is None:
+        meg = True
+        warn("The default of meg=True will change to meg=False in version "
+             "0.22.", DeprecationWarning)
```

Comment:
", set `meg` explicitly to avoid this warning"

... and you should only emit this if there are actually MEG channels present. Otherwise the value of `meg` does not matter and people should not care/be warned (e.g., when working with just EEG or fNIRS data the `meg=True` default won't matter)

Reply:
You don't want to warn if people use the default? If we only warn if there are MEG channels, then people won't know that something changed if they just happen to load a file without MEG channels.

I can do it, but this also makes the code a bit more complicated too.

Reply:
> You don't want to warn if people use the default?

Only want to warn them when it matters/changes behavior. This is what we usually do, and it cuts down on noise for people unaffected by the change.

> people won't know that something changed if they just happen to load a file without MEG channels.

This will only happen if someone uses a MEG+EEG file in 0.20 and got used to / expected that behavior, never use one in 0.21 (when they would get a warning), and then go back to using them again in 0.22, and they do not pay attention to the release notes. I say MEG+EEG because for MEG-only users, doing `pick_types(info, meg=False)` (default in 0.22) when you only have MEG channels things will either very clearly be wrong or you will get an error (I think).

This seems like an acceptable risk to take on given the benefits of not false-alarm notifying all our EEG, fNIRS, etc. users about things that are irrelevant to them.

Reply:
OK, then I'll try to keep the changes to issue the warning only in those cases as minimal as possible. Also, I think picking `meg=False` on a MEG-only data set will return an empty selection and not an error - at least this is the current behavior and I wouldn't change it.